### PR TITLE
p2p starknet 0.13.1

### LINKED ISSUE #1768: Starknet v0.13.1 P2P changes
The [protobufs for Starknet v0.13.1](https://github.com/starknet-io/starknet-p2p-specs/tree/shahak/preparation_for_0.13.1) are not quite ready yet. Once they are ready, our P2P layer needs to be updated accordingly. To realize the full scope of changes required, it might be a good idea to go through P2P changes in https://github.com/eqlabs/pathfinder/pull/1766 while implementing this issue and make sure that everything that was left out in https://github.com/eqlabs/pathfinder/pull/1766 gets implemented.

### PR DESCRIPTION
**Pull request by CHr15F0x**
Based on this spec [PR](https://github.com/starknet-io/starknet-p2p-specs/pull/29).

Fixes #1768.

### CODE REVIEW

File: crates/gateway-types/src/reply.rs [hunk 1]
```diff
@@ -2399,7 +2399,7 @@ mod tests {
             serde_json::from_str::<MaybePendingBlock>(integration::block::NUMBER_319709).unwrap();
             serde_json::from_str::<MaybePendingBlock>(v0_13_0::block::PENDING).unwrap();
             // This is from integration starknet_version 0.13.0 and contains data gas prices.
-            serde_json::from_str::<crate::reply::Block>(integration::block::NUMBER_329543).unwrap();
+            serde_json::from_str::<MaybePendingBlock>(integration::block::NUMBER_329543).unwrap();
```
Comment (kkovaacs) [inline crates/gateway-types/src/reply.rs:2402]: Uh, this hunk seems to be unrelated.

**Review by kkovaacs — APPROVED** [review]

File: crates/gateway-types/src/reply.rs (hunk 1 above)
Comment (CHr15F0x) [reply crates/gateway-types/src/reply.rs:2402]: I must have done this when rebasing, good catch!
